src/main/java/com/rybka/command/ConvertTask.java

```diff
@@ -0,0 +1,46 @@
+package com.rybka.command;
+
+import com.rybka.dao.CurrencyHistoryDAO;
+import com.rybka.model.CurrencyHistory;
+import com.rybka.service.exchange.ExchangeService;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j;
+
+@Log4j
+@AllArgsConstructor
+public class ConvertTask {
+    private final ExchangeService service;
+    private final CurrencyHistoryDAO currencyHistoryDAO;
+    private final String userBaseCurrency;
```

Comment:
How are you going to provide this values?

Reply:
fixed